aviatrix/resource_aviatrix_cloudn_edge_gateway.go

```diff
@@ -0,0 +1,300 @@
+	d.Set("management_interface_ip", cloudnEdgeGateway.ManagementInterfaceIp)
+	d.Set("default_gateway_ip", cloudnEdgeGateway.DefaultGatewayIp)
+	d.Set("dns_server", cloudnEdgeGateway.DnsServer)
+	d.Set("secondary_dns_server", cloudnEdgeGateway.SecondaryDnsServer)
+
+	if cloudnEdgeGateway.Dhcp {
+		d.Set("management_connection_type", "DHCP")
+	} else {
+		d.Set("management_connection_type", "Static")
+	}
+
+	gateway := &goaviatrix.TransitVpc{
+		GwName: d.Get("gw_name").(string),
+	}
+	transitGatewayAdvancedConfig, err := client.GetTransitGatewayAdvancedConfig(gateway)
+	if err != nil {
+		return diag.Errorf("failed to read cloudn edge gateway transit gateway advanced config: %v", err)
+	}
+	if transitGatewayAdvancedConfig.LocalASNumber != "" {
+		d.Set("local_as_number", transitGatewayAdvancedConfig.LocalASNumber)
+		d.Set("prepend_as_path", transitGatewayAdvancedConfig.PrependASPath)
+	}
+
+	d.SetId(cloudnEdgeGateway.GatewayName)
+	return nil
+}
+
+func resourceAviatrixCloudnEdgeGatewayUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	d.Partial(true)
+	gateway := &goaviatrix.TransitVpc{
+		GwName: d.Get("gw_name").(string),
+	}
+
+	if d.HasChanges("local_as_number", "prepend_as_path") {
+		var prependASPath []string
+		for _, v := range d.Get("prepend_as_path").([]interface{}) {
+			prependASPath = append(prependASPath, v.(string))
+		}
+
+		if (d.HasChange("local_as_number") && d.HasChange("prepend_as_path")) || len(prependASPath) == 0 {
+			// prependASPath must be deleted from the controller before local_as_number can be changed
+			// Handle the case where prependASPath is empty here so that the API is not called twice
+			err := client.SetPrependASPath(gateway, nil)
+			if err != nil {
+				return diag.Errorf("failed to delete prepend_as_path during Aviatrix CloudN Registration update: %v", err)
+			}
+		}
+
+		if d.HasChange("local_as_number") {
+			localASNumber := d.Get("local_as_number").(string)
+			err := client.SetLocalASNumber(gateway, localASNumber)
+			if err != nil {
+				return diag.Errorf("failed to update Aviatrix CloudN Registration: could not set local_as_number: %v", err)
+			}
+		}
+
+		if d.HasChange("prepend_as_path") && len(prependASPath) > 0 {
+			err := client.SetPrependASPath(gateway, prependASPath)
+			if err != nil {
+				return diag.Errorf("failed to update Aviatrix CloudN Registration prepend_as_path: %v", err)
+			}
+		}
+	}
+	d.Partial(false)
+
+	return resourceAviatrixCloudnEdgeGatewayRead(ctx, d, meta)
+}
+
+func resourceAviatrixCloudnEdgeGatewayDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	gwName := d.Get("gw_name").(string)
+	imageDownloadPath := d.Get("image_download_path").(string)
+
+	err := client.DeleteCloudnEdgeGateway(ctx, gwName)
+	if err != nil {
+		return diag.Errorf("could not delete cloudn edge gateway: %v", err)
```

Comment:
Should be consistent on how the error messages capitalize the resource. Below you use CloudN Edge Gateway.